addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3635,11 +3631,23 @@ const SnippetOptionWidget = Widget.extend({
 
                 _restoreTransitions();
 
-                const value = cssValues.join(' ');
+                let value = cssValues.join(' ');
+                if (params.withGradients && params.cssProperty === 'background-color') {
+                    // Check if there is a gradient, in that case this is the
+                    // value to be returned, we normally do not allow color and
+                    // gradient at the same time (the option would remove one
+                    // if editing the other).
+                    const parts = backgroundImageCssToParts(styles['background-image']);
+                    const hasTextGradientStyle = styles['-webkit-background-clip'] === 'text';
+                    if (parts.gradient && !hasTextGradientStyle) {
```

Comment:
Indeed, since the presets do not allow text gradients, this is not needed anymore.
No fix needed because this is not a real situation (unless people try to set up text gradients manually in custom CSS).